### PR TITLE
Include "special" _octodns values in Record.__repr__

### PR DESCRIPTION
**Pull request by ross**
This includes the `octodns` bits in `Record.__repr__`. The upside of this is that you can see the special bits that may be pushed to your providers, downside is that it could be noisy and not all special things apply to all providers and things pulled out of providers will only include the things that provider knows about.

## Example of the "good" case, CF proxied

```yaml
'*.sub':
  octodns:
    cloudflare:
      proxied: True
  ttl: 144
  type: A
  values: 
  - 2.2.3.3
  - 3.3.4.4
  - 4.4.5.5
```

```
********************************************************************************
* githubtest.net.
********************************************************************************
* cloudflare (CloudflareProvider)
*   Update
*     <ARecord A 144, *.sub.githubtest.net., ['2.2.3.3', '3.3.4.4', '4.4.5.5'], cloudflare=cloudflare.proxied=False> ->
*     <ARecord A 144, *.sub.githubtest.net., ['2.2.3.3', '3.3.4.4', '4.4.5.5'], cloudflare=cloudflare.proxied=True> (config)
*   Summary: Creates=0, Updates=1, Deletes=0, Existing Records=20
********************************************************************************
```

## Example of the "bad" case, where things aren't super clear

```yaml
'*.sub':
  octodns:
    cloudflare:
      other: False
      proxied: True
    something: True
  ttl: 144
  type: A
  values: 
  - 2.2.3.3
  - 3.3.4.4
  - 4.4.5.5
```

```
********************************************************************************
* githubtest.net.
********************************************************************************
* cloudflare (CloudflareProvider)
*   Update
*     <ARecord A 144, *.sub.githubtest.net., ['2.2.3.3', '3.3.4.4', '4.4.5.5'], cloudflare=cloudflare.proxied=False> ->
*     <ARecord A 144, *.sub.githubtest.net., ['2.2.3.3', '3.3.4.4', '4.4.5.5'], cloudflare=cloudflare.other=False, cloudflare.proxied=True, something=True> (config)
*   Summary: Creates=0, Updates=1, Deletes=0, Existing Records=20
********************************************************************************
```

/cc https://github.com/github/octodns/issues/284
/cc @parkr @begincalendar

### CODE REVIEW

**Comment by ross** [comment]
Btw, I'm not really sure about this PR. I'd like to show the special bits, but worried that it'll end up confusing things as much as helping...

File: octodns/record.py [hunk 1]
```diff
@@ -10,6 +10,16 @@
 import re
 
 
+def _stringify(d, prefix=''):
+    buf = []
+    for k, v in sorted(d.items()):
```
Comment (begincalendar) [inline octodns/record.py]: I think this produces a clearer result:
```python
        if isinstance(v, dict):
            buf.append(_stringify(v, '{}{}.'.format(prefix, k)))
        else:
            buf.append('{}{}={}'.format(prefix, k, v))
```

What do you think?

**Comment by begincalendar** [comment]
I think it gets the job done.
Which part don't you like?

**Comment by parkr** [comment]
I dig this! Here's an example:

```text
********************************************************************************
* parkermakes.tools.
********************************************************************************
* cloudflare (CloudflareProvider)
*   Update
*     <AliasRecord ALIAS 1, parkermakes.tools., kondwani.parkermakes.tools.> ->
*     <AliasRecord ALIAS 1, parkermakes.tools., kondwani.parkermakes.tools.> (config)
*   Summary: Creates=0, Updates=1, Deletes=0, Existing Records=15
********************************************************************************
* keltermoore.com.
********************************************************************************
* cloudflare (CloudflareProvider)
*   Update
*     <ARecord A 1, keltermoore.com., ['185.199.108.153'], cloudflare=cloudflare.proxied=True> ->
*     <ARecord A 1, keltermoore.com., ['185.199.108.153'], cloudflare=cloudflare.proxied=False> (config)
*   Summary: Creates=0, Updates=1, Deletes=0, Existing Records=2
********************************************************************************
```

**Comment by parkr** [comment]
Hm; the AliasRecord there wasn't changed by me, so looks like a diff is found there when nothing was changed?


The keltermoore.com change _was_ purposeful.

**Comment by parkr** [comment]
The AliasRecord diff doesn't get printed on `master`. Only:

```text
********************************************************************************
* keltermoore.com.
********************************************************************************
* cloudflare (CloudflareProvider)
*   Update
*     <ARecord A 1, keltermoore.com., ['185.199.108.153']> ->
*     <ARecord A 1, keltermoore.com., ['185.199.108.153']> (config)
*   Summary: Creates=0, Updates=1, Deletes=0, Existing Records=2
********************************************************************************
```

**Pull request closed without merging**